src/gui/qgsadvanceddigitizingfloater.h

```diff
@@ -0,0 +1,108 @@
+/***************************************************************************
+    qgsadvanceddigitizingfloater.cpp  -  floater for CAD tools
+    ----------------------
+    begin                : May 2019
+    copyright            : (C) Olivier Dalang
+    email                : olivier.dalang@gmail.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSADVANCEDDIGITIZINGFLOATER
+#define QGSADVANCEDDIGITIZINGFLOATER
+
+#include <QWidget>
+#include <QString>
+
+#include "ui_qgsadvanceddigitizingfloaterbase.h"
+#include "qgsadvanceddigitizingdockwidget.h"
+#include "qgis_gui.h"
+#include "qgis_sip.h"
+
+class QgsMapCanvas;
+class QgsAdvancedDigitizingDockWidget;
+
+/**
+* \ingroup gui
+* \brief The QgsAdvancedDigitizingFloater class is widget that floats
+* next to the mouse pointer, and allow interaction with the AdvancedDigitizing
+* feature. It proxies display and actions to QgsMapToolAdvancedDigitizingDockWidget.
+*/
+class GUI_EXPORT QgsAdvancedDigitizingFloater : public QWidget, private Ui::QgsAdvancedDigitizingFloaterBase
+{
+    Q_OBJECT
+
+  public:
+
+    explicit QgsAdvancedDigitizingFloater( QgsMapCanvas *canvas, QgsAdvancedDigitizingDockWidget *cadDockWidget );
+
+    /**
+    * Whether the floater is active or not.
+    * Note that the floater may be active but not visible (e.g. if the mouse is not over the canvas).
+    */
+    bool active();
+
+  public slots:
+
+    /**
+    * Set whether the floater should be active or not.
+    * Note that the floater may be active but not visible (e.g. if the mouse is not over the canvas).
+    *
+    * \param active
+    */
+    void setActive( bool active );
```

Comment:
I think that's never used as a slot (only directly called). Just make it a public method

Reply:
I'll refactor this a little bit (remove the toggleFloater function, and connect the action directly to floater::setActive). I think it's a bit cleaner (e.g. the settings to store the floater's visibility are now read/written in the floater class).